### ISSUE TITLE
Inefficient use of platform secrets API

### ISSUE THREAD

**Issue by PaulAnnekov** [elasticio/rest-api-component-v2#29]
# Internal Issue/Enhancement

## Description

For each incoming message in action or trigger you make an HTTP request to get a secret by ID.
https://github.com/elasticio/rest-api-component-v2/blob/ceab98fa1467562f844c102e6cb14e4d16555e83/lib/utils.js#L112

![giphy](https://user-images.githubusercontent.com/1203892/170213177-d0fe660e-272e-46d3-9666-4e790503e563.gif)

There are 2 problems with this approach:
1. You're abusing our API by making unnecessary API calls.
2. You make your component slower, because on each message it waits for reply from our API.

## Definition of Done

Use lazy load approach or load secret on component init. So you load secret only once and use it for the lifetime of the component.

**Comment by jhorbulyk** [comment]
Alternate approach: Move this logic into Sailor.

**Comment by PaulAnnekov** [comment]
> Alternate approach: Move this logic into Sailor.

Yep, it would be nice, but even if we will update sailor, this component will still need to be updated by removing current logic and adapting to new one. I think it's faster to fix it right here now and then do it later in sailor.
